diagnostics/factory.go

```diff
@@ -0,0 +1,22 @@
+package diagnostics
+
+import (
+	"context"
+	"io"
+
+	"github.com/treeverse/lakefs/block"
+	"github.com/treeverse/lakefs/catalog"
+	"github.com/treeverse/lakefs/config"
+	"github.com/treeverse/lakefs/db"
+)
+
+type Collector interface {
+	Collect(ctx context.Context, w io.Writer) (err error)
+}
+
+func CreateCollector(adb db.Database, cataloger catalog.Cataloger, cfg *config.Config, adapter block.Adapter) Collector {
```

Comment:
Not sure we need factory - we can have one collect implementation and return it.
The 'if' that checks what needs to collect and be inside the 'Collect' implementation.
WDYT?

Reply:
My main reason for doing it is to change the existing functionality as little as I could.
One design pattern or the other is not a big deal IMO, I will change if you insist.